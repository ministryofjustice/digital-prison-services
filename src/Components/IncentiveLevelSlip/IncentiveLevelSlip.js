import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { H2 } from '@govuk-react/heading'
import { BORDER_COLOUR } from 'govuk-colours'
import { spacing, typography } from '@govuk-react/lib'
import moment from 'moment'
import { forenameToInitial } from '../../utils'

const IncentiveLevelSlipDetails = styled.div`
  ${typography.font({ size: 19 })};
  border-bottom: 1px dashed ${BORDER_COLOUR};
`

const Row = styled.div`
  display: flex;
  align-items: center;
  border-bottom: ${(props) => (props.noBorder ? 'none' : `1px solid ${BORDER_COLOUR}`)};
  margin-bottom: ${spacing.simple(3)}px;
  padding-bottom: ${spacing.simple(3)}px;
`

const Column = styled.div`
  flex: 1 0 0;
  padding-right: ${spacing.simple(3)}px;
`

const Value = styled.span`
  font-weight: bold;
  display: ${(props) => (props.inline ? 'inline' : 'block')};
`

const Amendment = styled.div`
  margin-top: ${spacing.simple(3)}px;
`

function IncentiveLevelSlip({
  type,
  raisedDate,
  raisedBy,
  issuedBy,
  offenderName,
  offenderNo,
  caseNote,
  activityName,
  cellLocation,
  amendments,
}) {
  return (
    <IncentiveLevelSlipDetails>
      <Row>
        <Column>
          <H2 size="MEDIUM" mb={0}>
            {type || 'Incentive Level warning'}
          </H2>
        </Column>
        <Column>
          Date and time: <Value inline>{moment(raisedDate).format('D MMMM YYYY, h:mm a')}</Value>
        </Column>
      </Row>
      <Row>
        <Column>
          Name: <Value>{offenderName}</Value>
        </Column>
        <Column>
          Number: <Value>{offenderNo}</Value>
        </Column>
        <Column>
          Location: <Value>{cellLocation}</Value>
        </Column>
        {activityName && (
          <Column>
            Activity or area: <Value>{activityName}</Value>
          </Column>
        )}
      </Row>
      <Row>
        <Column>
          Reason:
          <Value>{caseNote}</Value>
          {amendments.length > 0 &&
            amendments.map((amendment) => (
              <Amendment key={amendment.creationDateTime}>
                Amendment: <Value>{amendment.additionalNoteText}</Value>
                {amendment.creationDateTime && (
                  <div>{moment(amendment.creationDateTime).format('D MMMM YYYY, h:mm a')}</div>
                )}
                {amendment.authorName && <div>{forenameToInitial(amendment.authorName)}</div>}
              </Amendment>
            ))}
        </Column>
      </Row>
      <Row noBorder>
        <Column>
          Raised by: <Value inline>{forenameToInitial(raisedBy)}</Value>
        </Column>
        <Column>
          Date: <Value inline>{moment(raisedDate).format('D MMMM YYYY')}</Value>
        </Column>
      </Row>
      <Row noBorder>
        <Column>
          Issued by: <Value inline>{forenameToInitial(issuedBy)}</Value>
        </Column>
        <Column>
          Date: <Value inline>{moment().format('D MMMM YYYY')}</Value>
        </Column>
      </Row>
    </IncentiveLevelSlipDetails>
  )
}

IncentiveLevelSlip.propTypes = {
  type: PropTypes.string,
  raisedDate: PropTypes.string,
  raisedBy: PropTypes.string,
  issuedBy: PropTypes.string,
  offenderName: PropTypes.string,
  offenderNo: PropTypes.string,
  caseNote: PropTypes.string,
  activityName: PropTypes.string,
  cellLocation: PropTypes.string,
  amendments: PropTypes.arrayOf(
    PropTypes.shape({
      authorName: PropTypes.string,
      creationDateTime: PropTypes.string,
      additionalNoteText: PropTypes.string,
    })
  ),
}

IncentiveLevelSlip.defaultProps = {
  type: undefined,
  raisedDate: undefined,
  raisedBy: undefined,
  issuedBy: undefined,
  offenderName: undefined,
  offenderNo: undefined,
  caseNote: undefined,
  activityName: undefined,
  cellLocation: undefined,
  amendments: [],
}

export default IncentiveLevelSlip
