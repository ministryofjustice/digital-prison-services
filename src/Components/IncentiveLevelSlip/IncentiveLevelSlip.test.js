import React from 'react'
import renderer from 'react-test-renderer'

import IncentiveLevelSlip from '.'

const props = {
  raisedBy: 'An Officer',
  offenderName: 'An Offender',
  offenderNo: 'ABC123',
  caseNote: 'Refused: Did not want to work',
  cellLocation: 'LEI-123',
}

describe('<IncentiveLevelSlip />', () => {
  Date.now = jest.fn(() => new Date(Date.UTC(2019, 0, 1)).valueOf())
  it('renders correctly', () => {
    const tree = renderer.create(<IncentiveLevelSlip {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with an activity or area', () => {
    const tree = renderer.create(<IncentiveLevelSlip {...props} activityName="Gym" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with an issued by name', () => {
    const tree = renderer
      .create(<IncentiveLevelSlip {...props} activityName="Gym" issuedBy="Another Officer" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with amendments', () => {
    const tree = renderer
      .create(
        <IncentiveLevelSlip
          {...props}
          amendments={[
            {
              authorName: 'Author, Amendment',
              creationDateTime: '2019-01-01T14:17:30',
              additionalNoteText: 'Rescinded',
            },
            { authorName: 'Amendment, Author', creationDateTime: '2019-01-01T15:17:30', additionalNoteText: 'Updated' },
          ]}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
