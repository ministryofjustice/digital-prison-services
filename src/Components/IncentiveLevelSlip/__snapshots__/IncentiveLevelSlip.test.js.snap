// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncentiveLevelSlip /> renders correctly 1`] = `
<div
  className="sc-bdVaJa kdwVCv"
>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      <h2
        className="heading__StyledHeading-sc-1sovo83-0 jdjFtq"
        size="MEDIUM"
      >
        Incentive Warning
      </h2>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date and time: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Name: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Number: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Location: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        LEI-123
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Reason:
      <span
        className="sc-bxivhb fFRjPw"
      >
        Refused: Did not want to work
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Raised by: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        A. Officer
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Issued by: 
      <span
        className="sc-bxivhb iPvUcL"
      />
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;

exports[`<IncentiveLevelSlip /> renders with amendments 1`] = `
<div
  className="sc-bdVaJa kdwVCv"
>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      <h2
        className="heading__StyledHeading-sc-1sovo83-0 jdjFtq"
        size="MEDIUM"
      >
        Incentive Warning
      </h2>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date and time: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Name: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Number: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Location: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        LEI-123
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Reason:
      <span
        className="sc-bxivhb fFRjPw"
      >
        Refused: Did not want to work
      </span>
      <div
        className="sc-ifAKCX dGVVEy"
      >
        Amendment: 
        <span
          className="sc-bxivhb fFRjPw"
        >
          Rescinded
        </span>
        <div>
          1 January 2019, 2:17 pm
        </div>
        <div>
          A. Amendment
        </div>
      </div>
      <div
        className="sc-ifAKCX dGVVEy"
      >
        Amendment: 
        <span
          className="sc-bxivhb fFRjPw"
        >
          Updated
        </span>
        <div>
          1 January 2019, 3:17 pm
        </div>
        <div>
          A. Author
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Raised by: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        A. Officer
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Issued by: 
      <span
        className="sc-bxivhb iPvUcL"
      />
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;

exports[`<IncentiveLevelSlip /> renders with an activity or area 1`] = `
<div
  className="sc-bdVaJa kdwVCv"
>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      <h2
        className="heading__StyledHeading-sc-1sovo83-0 jdjFtq"
        size="MEDIUM"
      >
        Incentive Warning
      </h2>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date and time: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Name: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Number: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Location: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        LEI-123
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Activity or area: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        Gym
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Reason:
      <span
        className="sc-bxivhb fFRjPw"
      >
        Refused: Did not want to work
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Raised by: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        A. Officer
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Issued by: 
      <span
        className="sc-bxivhb iPvUcL"
      />
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;

exports[`<IncentiveLevelSlip /> renders with an issued by name 1`] = `
<div
  className="sc-bdVaJa kdwVCv"
>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      <h2
        className="heading__StyledHeading-sc-1sovo83-0 jdjFtq"
        size="MEDIUM"
      >
        Incentive Warning
      </h2>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date and time: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019, 12:00 am
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Name: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        An Offender
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Number: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        ABC123
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Location: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        LEI-123
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Activity or area: 
      <span
        className="sc-bxivhb fFRjPw"
      >
        Gym
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH djUGzB"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Reason:
      <span
        className="sc-bxivhb fFRjPw"
      >
        Refused: Did not want to work
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Raised by: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        A. Officer
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
  <div
    className="sc-bwzfXH eHUHef"
  >
    <div
      className="sc-htpNat eOjbqm"
    >
      Issued by: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        A. Officer
      </span>
    </div>
    <div
      className="sc-htpNat eOjbqm"
    >
      Date: 
      <span
        className="sc-bxivhb iPvUcL"
      >
        1 January 2019
      </span>
    </div>
  </div>
</div>
`;
